src/bentoml/triton.py

```diff
@@ -0,0 +1,430 @@
+from __future__ import annotations
+
+import os
+import enum
+import shutil
+import typing as t
+import logging
+import urllib.parse
+from itertools import chain
+
+import fs
+import attr
+import fs.errors
+import fs.opener
+import fs.opener.errors
+from simple_di import inject
+from simple_di import Provide
+
+from .exceptions import StateException
+from ._internal.utils import bentoml_cattr
+from ._internal.utils import resolve_user_filepath
+from ._internal.runner.runner import RunnerMethod
+from ._internal.runner.runnable import RunnableMethodConfig
+from ._internal.runner.runner_handle import DummyRunnerHandle
+from ._internal.configuration.containers import BentoMLContainer
+from ._internal.runner.runner_handle.remote import TritonRunnerHandle
+
+if t.TYPE_CHECKING:
+    from fs.base import FS
+    from fs.opener.registry import Registry
+
+    from ._internal.runner.runner import Runner as BentoMLRunner
+    from ._internal.runner.runner_handle import RunnerHandle
+
+    P = t.ParamSpec("P")
+    R = t.TypeVar("R")
+
+    LogFormat = t.Literal["default", "ISO8601"]
+    GrpcInferResponseCompressionLevel = t.Literal["none", "low", "medium", "high"]
+    TraceLevel = t.Literal["OFF", "TIMESTAMPS", "TENSORS"]
+else:
+    LogFormat = str
+    GrpcInferResponseCompressionLevel = str
+    TraceLevel = str
+
+logger = logging.getLogger(__name__)
+
+__all__ = ["Runner"]
+
+_object_setattr = object.__setattr__
+
+
+@attr.frozen(slots=False, eq=False)
+class TritonRunner:
+    name: str
+    repository_path: str
+    models: list[str]
+    runnable_class: type[TritonRunner]
+
+    _fs: FS
+    _runner_handle: RunnerHandle = attr.field(init=False, factory=DummyRunnerHandle)
+
+    def __init__(
+        self,
+        name: str,
+        model_repository: str,
+        *,
+        fs_protocol: str | None = None,
+    ):
+        lname = name.lower()
+        if name != lname:
+            logger.warning(
+                "Converting runner name '%s' to lowercase: '%s'", name, lname
+            )
+
+        # TODO: Support configuration (P1)
+
+        # The logic below mimic the behaviour of Exportable.import_from
+        # To determine models inside given model repository.
+        try:
+            parsed_model_repository = fs.opener.parse(model_repository)
```

Comment:
Hm, why are we using `fs` here? I thought we were just passing this through to triton?

Reply:
we need to find out a way to assign the model inside the store as attributes to the runner.